### ISSUE TITLE
Rare crash after interacting with Torch UI

### ISSUE THREAD

**Issue by susu** [TorchAPI/Torch#152]
- [x] Verify that this issue is related to Torch and not a Torch plugin or the vanilla game
- [ ] Ensure that the issue is reproducible for testing (provide a link to a test world if necessary)
- [ ] Is this a suggestion?

**Torch Version:** 1.2.0.0 **SE Version:** 1.184.500

### Expected Behavior
That it is not crashing during I clicking around Torch UI

### Observed Behavior
It is crashing when I was clicking around Torch UI

### Steps to Reproduce
I'm not exactly sure, but it is somehow related with the Torch UI, either chat or entities list.
I will tell what I've done:
1. A player on my server was using that "rotor head factory" cheat with a timer block and a grinder.
2. I turned off the timer block from Torch UI / entities list
3. Then I warned them (via Torch UI chat) to stop abusing that cheat
4. Poor guy started to grind down his rotor and then his grinder, then server crash.

(I had another, earlier crash a day ago, but I don't remember what I was doing. It was also some clicking around in TorchUI, then crash)

### Other Information
Plugin: Essentials

I had two crash, their backtrace is similar, and I think is not related to SE code

```

18:25:58.3672 [FATAL] Initializer: System.InvalidOperationException: Added item does not appear at given index '12'.
  at MS.Internal.Data.EnumerableCollectionView.ProcessCollectionChanged(NotifyCollectionChangedEventArgs args)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   at System.Windows.Window.ShowHelper(Object booleanBox)
   at System.Windows.Window.ShowDialog()
   at Torch.Server.Initializer.Run() in C:\Program Files (x86)\Jenkins\workspace\Torch_Torch_staging-ZNS7FN4RGURQGN7YDLUHQ3UMYIBPHQSIJWKCAY6N7MCURWHGXRKA\Torch.Server\Initializer.cs:line 92
```
The previous crash:
```
22:23:01.0958 [FATAL] Initializer: System.InvalidOperationException: Added item does not appear at given index '8'.
   at MS.Internal.Data.EnumerableCollectionView.ProcessCollectionChanged(NotifyCollectionChangedEventArgs args)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   at System.Windows.Window.ShowHelper(Object booleanBox)
   at System.Windows.Window.ShowDialog()
   at Torch.Server.Initializer.Run() in C:\Program Files (x86)\Jenkins\workspace\Torch_Torch_staging-ZNS7FN4RGURQGN7YDLUHQ3UMYIBPHQSIJWKCAY6N7MCURWHGXRKA\Torch.Server\Initializer.cs:line 92
   at Torch.Server.Program.Main(String[] args) in C:\Program Files (x86)\Jenkins\workspace\Torch_Torch_staging-ZNS7FN4RGURQGN7YDLUHQ3UMYIBPHQSIJWKCAY6N7MCURWHGXRKA\Torch.Server\Program.cs:line 62

```

**Comment by Equinox-** [comment]
Should be fixed after 6f5142393b14057d2d929d6d1c3f651d89a88649

**Comment by Equinox-** [comment]
Have not seen this error in a long while, marking closed.
